boot/freeldr/freeldr/arch/i386/hwdisk.c

```diff
@@ -139,10 +139,24 @@ DiskRead(ULONG FileId, VOID* Buffer, ULONG N, ULONG* Count)
     BOOLEAN ret;
     ULONGLONG SectorOffset;
 
+    if (DiskReadBufferSize == 0)
+    {
+        WARN("DiskRead(): DiskReadBufferSize is 0, something is wrong\n");
+        *Count = 0;
+        return ENOMEM;
+    }
+
     TotalSectors = (N + Context->SectorSize - 1) / Context->SectorSize;
     MaxSectors   = DiskReadBufferSize / Context->SectorSize;
     SectorOffset = Context->SectorNumber + Context->SectorOffset;
 
+    if (MaxSectors == 0)
```

Comment:
MaxSectors will be 0 only when DiskReadBufferSize is 0. I think some static analyzer can even tell you that this is a dead code :)